crates/cadenza/src/repl.rs

```diff
@@ -0,0 +1,310 @@
+//! REPL (Read-Eval-Print Loop) for Cadenza.
+//!
+//! Provides an interactive environment with:
+//! - Command history (saved to ~/.cadenza_history)
+//! - Syntax highlighting
+//! - Auto-completion for identifiers
+//! - Option to load files into scope
+
+use anyhow::Result;
+use cadenza_eval::{Compiler, Env, Value};
+use cadenza_syntax::{lexer::Lexer, parse::parse, token::Kind};
+use rustyline::{
+    Context, Editor, Helper,
+    completion::{Completer, Pair},
+    error::ReadlineError,
+    highlight::{CmdKind, Highlighter},
+    hint::Hinter,
+    validate::Validator,
+};
+use std::{borrow::Cow, path::PathBuf};
+
+/// REPL helper that provides completion and syntax highlighting
+struct CadenzaHelper;
+
+impl CadenzaHelper {
+    fn new() -> Self {
+        Self
+    }
+}
+
+impl Helper for CadenzaHelper {}
+
+impl Completer for CadenzaHelper {
+    type Candidate = Pair;
+
+    fn complete(
+        &self,
+        line: &str,
+        pos: usize,
+        _ctx: &Context<'_>,
+    ) -> rustyline::Result<(usize, Vec<Pair>)> {
+        // Find the start of the current word
+        let start = line[..pos]
+            .rfind(|c: char| !c.is_alphanumeric() && c != '_' && c != '-')
+            .map(|i| i + 1)
+            .unwrap_or(0);
+
+        let word = &line[start..pos];
+        if word.is_empty() {
+            return Ok((pos, Vec::new()));
+        }
+
+        // Collect matching identifiers from the environment
+        let mut candidates = Vec::new();
+
+        // Get identifiers from the environment
+        // Note: Env doesn't expose a way to iterate over bindings,
+        // so we'll provide a basic set of built-in names
+        let builtins = [
+            "let", "fn", "=", "match", "assert", "typeof", "measure", "+", "-", "*", "/", "==",
+            "!=", "<", "<=", ">", ">=", "|>",
+        ];
```

Comment:
We should provide a way to do this. At the very least we need a STATUS.md document to track what gaps we have in functionality and potential enhancements/features.